provider/aws/metrics.go

```diff
@@ -23,6 +23,10 @@ var (
 )
 
 func (p *Provider) cloudwatchMetrics(mdqs []metricDataQuerier, opts structs.MetricsOptions) (structs.Metrics, error) {
+	if len(mdqs) == 0 {
```

Comment:
is the current behavior failing somewhere?
can't we just return an empty result instead of the error?

Reply:
> is the current behavior failing somewhere?
It fails on [ProcessList](https://github.com/convox/rack/blob/58cea6e71221f0883f7708224d67cbd9083ef3f0/provider/aws/processes.go) which is used by the `convox ps`, `convox exec` and `convox run`.

> can't we just return an empty result instead of the error?

My opinion is that returning an error can be more helpful in cases where it should return metrics and it's not returning.